File: frontend/src/types/auditLog.ts
export type AuditAction = 0 | 1 | 2 | 3 | 4;

export type AuditLog = {
  id: number;
  entityName: string;
  entityId: string;
  action: AuditAction;
  userId: number | null;
  username: string;
  timestamp: string;
  oldValues: string | null;
  newValues: string | null;
  ipAddress: string | null;
};

import type { PagedResult } from './paged';

export type AuditLogPage = PagedResult<AuditLog>;

export type AuditLogFilter = {
  entityName?: string;
  userId?: number;
  action?: AuditAction;
  dateFrom?: string;
  dateTo?: string;
  page: number;
  pageSize: number;
};

File: backend/src/TMS.Domain/Entities/AuditLog.cs
using TMS.Domain.Enums;

namespace TMS.Domain.Entities
{
    public class AuditLog
    {
        public int Id { get; set; }
        public string EntityName { get; set; } = string.Empty;
        public string EntityId { get; set; } = string.Empty;
        public AuditAction Action { get; set; }
        public int? UserId { get; set; }
        public string Username { get; set; } = string.Empty;
        public DateTime Timestamp { get; set; } = DateTime.UtcNow;
        public string? OldValues { get; set; }
        public string? NewValues { get; set; }
        public string? IpAddress { get; set; }
    }
}